Python3 support

olf42 opened the pull request:
I made some modifications s.t. this app runs with python3

* print statements are now print functions
* use io instead of cstringIO
* encode strings as UTF-8 where bytes expected
* In CherryPy config set 'tools.encode.text_only': False to prevent ValueError: Page handlers MUST return bytes

olf42 committed:
Make transition to Python3

This updates Python to version 3.7.3

cfra reviewed (changes requested):
This seems to break the layer renderer.
There are quite some users which use the docker image like this:

```Dockerfile
FROM cfra/eventmap
COPY layers/ /srv/eventmap/layers/
RUN python read_layers.py
```

With the suggested changes, this breaks:

```console
Sending build context to Docker daemon  82.83MB
Step 1/16 : FROM eventmap-python3
 ---> 0a90a92b6d01
Step 2/16 : COPY layers/ /srv/eventmap/layers/
 ---> fc62942be4ef
Step 3/16 : RUN python read_layers.py
 ---> Running in 61f0455e819c
Traceback (most recent call last):
  File "read_layers.py", line 10, in <module>
    import cairo
ModuleNotFoundError: No module named 'cairo'
The command '/bin/sh -c python read_layers.py' returned a non-zero code: 1
```

cfra reviewed (changes requested):
Thank you for working on fixing the building of tiles again. :bowing_woman: 

It seems like it's not completely there yet, tough:

```
Traceback (most recent call last):
  File "read_layers.py", line 242, in <module>
    LayerInfoStore(layers).store(layer_info_path)
  File "read_layers.py", line 215, in store
    for layer in sorted(self.layers):
TypeError: '<' not supported between instances of 'PdfLayer' and 'PdfLayer'
```

Using the `debian:buster` base image instead of the Python one also changed the interpreters name from `python` to `python3` though, so the `CMD` part of the `Dockerfile` probably needs to updated.

When I try to run the newly built image, I get the following right now:

```console
docker run --rm -ti cfra/eventmap
docker: Error response from daemon: OCI runtime create failed: container_linux.go:346: starting container process caused "exec: \"python\": executable file not found in $PATH": unknown.
```

cfra committed:
Fix layer ordering

__cmp__ is not respected in Python 3, therefore a TypeError is
raised in L215 in read_layers.py when it is attempted to sort
the layers.

There doesn't seem any necessity for layers being an ordered data type,
so make it explicit we are sorting the layers by name and drop
the comparison function from layers alltogether.

cfra reviewed (approved):
Thank you for performing this long overdue upgrade from Python2 to Python3. :bowing_woman: 

I have updated the sorting of layers in `read_layers.py` and I think everything is fine now. :sparkles:
